server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -455,4 +558,44 @@ export class TenantManager {
         const accessInfo = JSON.parse(this.secretManager.decryptSecret(encryptedAccessInfo));
         return accessInfo;
     }
+
+    private decryptCachedKeys(cachedKey: string) {
+        const keys = JSON.parse(cachedKey);
+        return keys.key2 === "" ? {
+                key1: this.secretManager.decryptSecret(keys.key1),
+                key2: "",
+            } : {
+                key1: this.secretManager.decryptSecret(keys.key1),
+                key2: this.secretManager.decryptSecret(keys.key2),
+            };
+    }
+
+    private async getCacheKey(tenantId: string): Promise<string> {
+        return this.cache?.get(`tenantKeys:${tenantId}`);
+    }
+
+    private async deleteCacheKey(tenantId: string): Promise<boolean> {
+        return this.cache?.delete(`tenantKeys:${tenantId}`);
+    }
+
+    private async setCacheKey(tenantId: string, value: ITenantKeys, metric?: Lumber<LumberEventName.RiddlerFetchTenantKey>): Promise<void> {
+        const lumberProperties = { [BaseTelemetryProperties.tenantId]: tenantId };
+        try {
+            await this.cache?.set(`tenantKeys:${tenantId}`, JSON.stringify(value));
+            Lumberjack.info(`Added new tenant keys to cache.`, lumberProperties);
+        } catch (error) {
+            if (metric) {
+                metric.setProperty("CacheFailed", true);
```

Comment:
`CacheFailed` seems a bit vague without the context. Can we rename it to something like `"Setting key in cache failed"`?